packages/Manager/src/test/kotlin/nz/govt/eop/consumers/WaterAllocationConsumerTest.kt

```diff
@@ -31,63 +33,73 @@ class WaterAllocationConsumerTest(@Autowired val context: DSLContext) {
   @Test
   fun `Should create an allocation if it does not exist`() {
     // GIVEN
-    val message =
-        WaterAllocationMessage("area-id-create", BigDecimal("100.11"), "ingest-id", Instant.now())
+    val message = createWaterAllocationMessage()
 
     // WHEN
     consumer.processMessage(message)
 
     // THEN
-    val records = fetchWaterAllocations(message.areaId)
+    val records = fetchWaterAllocations(message.sourceId)
     records.size.shouldBe(1)
     val record = records.first()
-    record.amount.shouldBe(message.amount)
-    record.lastUpdatedIngestId.shouldBe(message.ingestId)
+    record.allocation.shouldBe(message.allocation)
+    record.ingestId.shouldBe(message.ingestId)
   }
 
   @Test
-  fun `Should update an allocations if exists`() {
+  fun `Should create a new allocation with updated data`() {
     // GIVEN
-    val firstMessage =
-        WaterAllocationMessage("area-id-update", BigDecimal("100.11"), "ingest-id-1", Instant.now())
+    //   val now = OffsetDateTime.now(ZoneOffset.UTC)
+    //   val nowPlus2Hours = now.plus(Duration.ofHours(2))
+    val firstMessage = createWaterAllocationMessage()
     val secondMessage =
-        WaterAllocationMessage("area-id-update", BigDecimal("200.22"), "ingest-id-2", Instant.now())
+        firstMessage.copy(allocation = BigDecimal("200.11"), ingestId = "secondIngestId")
 
     // WHEN
     consumer.processMessage(firstMessage)
     consumer.processMessage(secondMessage)
 
     // THEN
-    val records = fetchWaterAllocations(firstMessage.areaId)
-    records.size.shouldBe(1)
-    val record = records.first()
-    record.amount.shouldBe(secondMessage.amount)
-    record.lastUpdatedIngestId.shouldBe(secondMessage.ingestId)
-  }
-
-  @Test
-  fun `Should not update an allocations if older data received `() {
-    // GIVEN
-    val firstMessage =
-        WaterAllocationMessage(
-            "area-id-no-update", BigDecimal("100.11"), "ingest-id-1", Instant.now())
-    val yesterday = Instant.now().minus(1, ChronoUnit.DAYS)
-    val secondMessage =
-        WaterAllocationMessage("area-id-no-update", BigDecimal("200.22"), "ingest-id-2", yesterday)
+    val records = fetchWaterAllocations(firstMessage.sourceId)
+    records.size.shouldBe(2)
 
-    // WHEN
-    consumer.processMessage(firstMessage)
-    consumer.processMessage(secondMessage)
+    val firstRecord = records.first()
+    firstRecord.allocation.shouldBe(firstRecord.allocation)
+    firstRecord.ingestId.shouldBe(firstRecord.ingestId)
+    firstRecord.effectiveFrom?.toInstant().shouldBe(firstMessage.receivedAt)
+    firstRecord.effectiveTo
```

Comment:
@smozely: Is there a simpler way to do this? The truncation is needed to avoid the test [failing due to precision differences when reading a timestamp back from the db](https://github.com/Greater-Wellington-Regional-Council/Environmental-Outcomes-Platform/actions/runs/6154836688/job/16700809405?pr=92#step:5:246).

Reply:
Oh man, yeah I hate that thing of DB returning different precision, it bites me in every project somehow!

Couple of options ... maybe a util function to that takes an offset date time and returns an instant.

Being Kotlin you can do extension functions, which I generally hate (I've seen them get overused in other languages) ... but I like them for stuff like this where we can make tests more readable 

i.e. this 
`fun OffsetDateTime.truncateToMillis(): Instant = this.truncatedTo(ChronoUnit.MILLIS).toInstant()`

would let you do this ... 
`secondRecord.effectiveFrom!!.truncateToMillis().shouldBe(secondRecordReceivedAtTruncated)`

(I'd do !! here to force it to fail if effectiveFrom ended up being null ... I think the `?` would have just carried on and you wouldn't notice)


Otherwise there are a couple of other Assertions, i.e. `shouldBeCloseTo` 
`firstRecord.effectiveTo!!.toInstant().shouldBeCloseTo(secondRecordReceivedAtTruncated, Duration.ofSeconds(1))`

... but that kinda implies that they might not be exactly the same which isn't really what your after.